src/command_line_parser.cc

```diff
@@ -96,44 +97,87 @@ namespace triton { namespace server {
 // [FIXME] expose following parse helpers for other type of parser
 namespace {
 
+// A wrapper around std::stoi, std::stoull, std::stoll, std::stod
+// to catch `invalid argument` and `out of range` exceptions
+template <typename T>
+T
+SafeConvertion(
+    const std::string& arg, std::function<T(const std::string&)> stox,
+    const std::string& opt_name = "")
+{
+  try {
+    return stox(arg);
```

Comment:
One thing you might be able to do is use templates for stoxa and then could get rid of the function argument - and just have ParseOption be safeconversion...  Would something like this work?

```
template <typename T>
T stox(const std::string);

template<>
int stox();

template <typename T>
T ParseOption(string, name) {
  try {
      return stox<T>(arg) ;
} catch () {
}
}
```

Reply:
I am not sure I understand how it will work with `stod`, `stoull` and `stoll` as well

Reply:
I might be missing something here - but was thinking you could add template instantiations for `stox()` for each -

Reply:
Ah, so instead of bindings instantiate templates for `stox` ? That wasn't clear for me initially

Reply:
yep - if it works. I liked the way you used templates here and then just got to thinking if it could be used instead of bindings ...

Reply:
Sounds good, I'll try this out